### PR TITLE
Feat/llm flexible content cards

### PR DESCRIPTION
**Pull request by cgrellard-ledger**
### 📝 Description

New Flexible Content Cards System
Horizontal Card added
Carousel Layout added

https://github.com/LedgerHQ/ledger-live/assets/17146928/e0a20d68-47e4-45d0-9709-d1f14d8cd03f

### ❓ Context

- **JIRA or GitHub link**: [LIVE-9058] [LIVE-9793] [LIVE-10076] [LIVE-10606]

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - 

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-9058]: https://ledgerhq.atlassian.net/browse/LIVE-9058?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-9793]: https://ledgerhq.atlassian.net/browse/LIVE-9793?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIVE-10076]: https://ledgerhq.atlassian.net/browse/LIVE-10076?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[LIVE-10606]: https://ledgerhq.atlassian.net/browse/LIVE-10606?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/reducers/types.ts
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/reducers/types.ts:1]: Seems that you have Prettier errors

File: apps/ledger-live-mobile/src/dynamicContent/useDynamicContentLogic.ts [hunk 1]
```diff
@@ -46,6 +50,12 @@ export const useDynamicContentLogic = () => {
...
 
+    const categoriesCards = filterByType(mobileContentCards, ContentCardsType.category)
+      .map(card => mapAsCategoryContentCard(card))
+      .sort(compareCards);
+
+    dispatch(setDynamicContentCategoriesCards(categoriesCards));
+    dispatch(setDynamicContentMobileCards(mobileContentCards));
     dispatch(setDynamicContentWalletCards(walletCards));
     dispatch(setDynamicContentAssetsCards(assetCards));
     dispatch(setDynamicContentNotificationCards(notificationCards));
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/dynamicContent/useDynamicContentLogic.ts:61]: We have too much dispatch there + we should avoid to parse the data before storing it in the Redux Store. The flow should be:
`Fetch -> Dispatch -> Store -> Selector -> View`
Here we already prepare the data for our views or logic but it creates complexity if we want to do something else with the data we got from the API.

File: apps/ledger-live-mobile/src/dynamicContent/ContentCardsLocation.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  if (categoriesToDisplay.length === 0) return null;
+
+  const categoriesSorted = categoriesToDisplay.sort(compareCards);
+  const categoriesWithCards = categoriesSorted.map(category => ({
+    category,
+    cards: mobileCards.filter(mobileCard => mobileCard.extras.categoryId === category.categoryId),
+  }));
+  const categoriesWithAtLeastOneCard = categoriesWithCards.filter(
+    categoryWithCards => categoryWithCards.cards.length > 0,
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/dynamicContent/ContentCardsLocation.tsx:42]: As at the end, we only use `categoriesWithAtLeastOneCard` we could create a function to parse categoriesCards outside this component. That could even be covered by unit test.

**Review by KVNLS — CHANGES_REQUESTED** [review]

File: apps/ledger-live-mobile/src/contentCards/layouts/carousel/pagination/utilts.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ * Returns the status of an indexed item from the carousel index.
+ */
+export const getItemStatus = (itemIndex: number, carouselIndex: number) => {
+  return itemIndex === carouselIndex
+    ? ItemStatus.active
+    : Math.abs(itemIndex - carouselIndex) === 1
+    ? ItemStatus.nearby
+    : Math.abs(itemIndex - carouselIndex) === 2
+    ? ItemStatus.far
+    : ItemStatus.none;
```
[the remark this answers is not in the document]
Comment (sshmaxime) [reply apps/ledger-live-mobile/src/contentCards/layouts/carousel/pagination/utilts.tsx]: What would you suggest then ? I feel like that way it's somehow pretty easily readable, no ? wdyt @mcayuelas-ledger

File: apps/ledger-live-mobile/src/dynamicContent/useDynamicContentLogic.ts (hunk 1 above)
Comment (cgrellard-ledger) [reply apps/ledger-live-mobile/src/dynamicContent/useDynamicContentLogic.ts:61]: We'll implement this change in another PR as it implies some modifications on the existing content cards system

File: apps/ledger-live-mobile/src/contentCards/layouts/carousel/pagination/bullets.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import Animated, { FadeOut, Layout, useAnimatedStyle, withSpring } from "react-native-reanimated";
+import { useTheme } from "styled-components/native";
+import { ItemStatus } from "./types";
+
+const useBulletStyles = () => {
+  const { colors } = useTheme();
+
+  /* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/contentCards/layouts/carousel/pagination/bullets.tsx:9]: you can type this i think :)

**Review by KVNLS — APPROVED** [review]

**Pull request merged**